### PR TITLE
[Gallery] Add animation for settings backdrop

### PR DESCRIPTION
**Pull request by perclasson**
### Changes
Adding animation to settings backdrop, includes the following changes:
- Lift up AnimationController to parent widget AnimatedBackdrop so it can be shared with child widgets.
- Make the boolean 'isSettingsOpen' a ValueNotifier so we can listen to changes for it in SettingsPage.
- When closing settings, also shrink expanded setting.
- Animate the settings page to slide in from above.
- Animate the settings item to stagger when the panel is opened.

### Animation after change
See gif for animation:
settings screen animation

### Animation before change
settings screen animation before

### Desktop after (there is no change on desktop)
desktop after change

### Desktop before
desktop before change

### Closes
https://github.com/material-components/material-components-flutter-gallery/issues/24

### CODE REVIEW

File: gallery/gallery/lib/pages/backdrop.dart [hunk 1]
```diff
@@ -13,17 +13,92 @@ import 'package:gallery/constants.dart';
...
+  _AnimatedBackdropState createState() => _AnimatedBackdropState();
+}
+
+class _AnimatedBackdropState extends State<AnimatedBackdrop>
+    with SingleTickerProviderStateMixin {
+  AnimationController backdropController;
+  ValueNotifier<bool> isSettingsOpenNotifier;
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (rami-a) [inline gallery/gallery/lib/pages/backdrop.dart]: Nit: Put the `build` method last in the class

File: gallery/gallery/lib/pages/backdrop.dart [hunk 2]
```diff
@@ -13,17 +13,92 @@ import 'package:gallery/constants.dart';
...
+    final openSettingsAnimation = CurvedAnimation(
```
Comment (rami-a) [inline gallery/gallery/lib/pages/backdrop.dart]: Should these 2 animations not be initialized in `initState` instead?

File: gallery/gallery/lib/pages/settings.dart [hunk 3]
```diff
@@ -119,156 +139,168 @@ class _SettingsPageState extends State<SettingsPage> {
...
-              SettingsListItem<TargetPlatform>(
-                title:
-                    GalleryLocalizations.of(context).settingsPlatformMechanics,
-                selectedOption: options.platform,
-                options: LinkedHashMap.of({
-                  TargetPlatform.android: DisplayOption(
-                    GalleryLocalizations.of(context).settingsPlatformAndroid,
+                if (isDesktop)
+                  ...settingsListItems
+                else ...[
```
Comment (rami-a) [inline gallery/gallery/lib/pages/settings.dart]: Whoa, I had no idea you could do `else` statements

File: gallery/gallery/lib/pages/settings.dart [hunk 4]
```diff
@@ -383,3 +415,93 @@ class _SettingsLink extends StatelessWidget {
...
+              child: child,
+            ),
+          ],
+        );
+      });
+    }
+  }
+}
+
+/// Animte the settings list items to stagger in from above.
```
Comment (rami-a) [inline gallery/gallery/lib/pages/settings.dart]: Animate*

File: gallery/gallery/lib/pages/backdrop.dart (hunk 1 above)
Comment (perclasson) [reply gallery/gallery/lib/pages/backdrop.dart]: Done.

File: gallery/gallery/lib/pages/backdrop.dart (hunk 2 above)
Comment (perclasson) [reply gallery/gallery/lib/pages/backdrop.dart]: That works too, I'll do that.

**Pull request merged**